services/app-web/src/components/HealthPlanPackageTable/HealthPlanPackageTable.tsx

```diff
@@ -0,0 +1,333 @@
+import React from 'react'
+import {
+    ColumnFiltersState,
+    createColumnHelper,
+    flexRender,
+    getCoreRowModel,
+    getFilteredRowModel,
+    getSortedRowModel,
+    RowData,
+    useReactTable,
+    getFacetedUniqueValues,
+} from '@tanstack/react-table'
+import { HealthPlanPackageStatus, Program } from '../../gen/gqlClient'
+import styles from './HealthPlanPackageTable.module.scss'
+import { Table, Tag, Link } from '@trussworks/react-uswds'
+import { NavLink } from 'react-router-dom'
+import dayjs from 'dayjs'
+import classnames from 'classnames'
+import { SubmissionStatusRecord } from '../../constants/healthPlanPackages'
+import { FilterAccordion, FilterSelect } from '../FilterAccordion'
+import statePrograms from '../../common-code/data/statePrograms.json'
+
+declare module '@tanstack/table-core' {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    interface ColumnMeta<TData extends RowData, TValue> {
+        dataTestID: string
+    }
+}
+
+export type PackageInDashboardType = {
+    id: string
+    name: string
+    submittedAt?: string
+    updatedAt: Date
+    status: HealthPlanPackageStatus
+    programs: Program[]
+    submissionType?: string
+    stateName?: string
+}
+
+type UserType = 'StateUser' | 'CMSUser'
+
+export type PackageTableProps = {
+    tableData: PackageInDashboardType[]
+    userType: UserType
+    showFilters?: boolean
+}
+
+const isSubmitted = (status: HealthPlanPackageStatus) =>
+    status === 'SUBMITTED' || status === 'RESUBMITTED'
+
+function submissionURL(
+    id: PackageInDashboardType['id'],
+    status: PackageInDashboardType['status'],
+    userType: UserType
+): string {
+    if (userType === 'CMSUser') {
+        return `/submissions/${id}`
+    } else if (status === 'DRAFT') {
+        return `/submissions/${id}/edit/type`
+    } else if (status === 'UNLOCKED') {
+        return `/submissions/${id}/edit/review-and-submit`
+    }
+    return `/submissions/${id}`
+}
+
+const StatusTag = ({
+    status,
+}: {
+    status: HealthPlanPackageStatus
+}): React.ReactElement => {
+    const tagStyles = classnames('', {
+        [styles.submittedTag]: isSubmitted(status),
+        [styles.draftTag]: status === 'DRAFT',
+        [styles.unlockedTag]: status === 'UNLOCKED',
+    })
+
+    const statusText = isSubmitted(status)
+        ? SubmissionStatusRecord.SUBMITTED
+        : SubmissionStatusRecord[status]
+
+    return <Tag className={tagStyles}>{statusText}</Tag>
```

Comment:
todo: Use the new shared `<InfoTag />` component. This will also change the color slightly of Unlocked which is intended, the color it is right now is off from what is in Figma. The program tag and the unlocked tag are supposed to be same color